### PR TITLE
SDI-126: ⚰️  Remove old unused apis

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AbsentReasonsController.kt [hunk 1]
```diff
@@ -24,23 +23,17 @@ import uk.gov.justice.digital.hmpps.whereabouts.model.AbsentSubReason.Companion.
...
-  fun reasonsV2(): AbsentReasonsV2Dto =
-    AbsentReasonsV2Dto(
-      paidReasons.toReasonDto(),
-      unpaidReasons.toReasonDto(),
-      iepTriggers.toList(),
-      absentSubReasonTriggers,
-      paidSubReasons.toSubReasonDto(),
-      unpaidSubReasons.toSubReasonDto(),
-    )
+  fun reasonsV2() = reasons()
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AbsentReasonsController.kt:36]: Old `reasons()` version no longer used.  Will get front end to switch back from `/v2` version and then can remove `/v2` later on

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/attendance/AbsentReasonsDto.kt [hunk 2]
```diff
@@ -2,18 +2,34 @@ package uk.gov.justice.digital.hmpps.whereabouts.dto.attendance
 
 import io.swagger.annotations.ApiModelProperty
 import uk.gov.justice.digital.hmpps.whereabouts.model.AbsentReason
+import uk.gov.justice.digital.hmpps.whereabouts.model.AbsentSubReason
 
 data class AbsentReasonsDto(
   @ApiModelProperty(value = "List of paid absent reasons", example = "[\"ApprovedCourse\"]", position = 1)
-  val paidReasons: Set<AbsentReason>,
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/dto/attendance/AbsentReasonsDto.kt:8]: this dto replaced with the v2 version

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceStatistics.kt [hunk 3]
```diff
@@ -20,8 +20,6 @@ import java.time.LocalDate
 
 @Suppress("unused")
 data class PaidReasons(
-  @Deprecated("Moved up a level out of paid reasons")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceStatistics.kt:23]: remove deprecated stat here as front end now using higher level version

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**